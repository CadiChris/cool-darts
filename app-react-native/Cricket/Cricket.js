import React from 'react'
import TableauDesScores from './TableauDesScores/TableauDesScores'
import { StyleSheet, Text, View } from 'react-native'

import Partie from './Partie/Partie'

export default class App extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <Partie style={{height: '40%'}}/>
        <TableauDesScores />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#aaa',
    paddingTop: 25
  },
});
