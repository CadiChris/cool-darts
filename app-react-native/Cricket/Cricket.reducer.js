import { scoreVierge } from './TableauDesScores/arbitrage/score'
import { calculerLeNouveauScore } from './TableauDesScores/arbitrage/arbitre'
import { vainqueurs } from './TableauDesScores/arbitrage/vainqueurs'

const STATE_INITIAL = {
  phase: 'INSCRIPTION',
  joueurs: [],
  peutDemarrer: false,
  scores: [],
  vainqueurs: []
}

export default function partie(state = STATE_INITIAL, action) {
  switch (action.type) {
    case 'INSCRIRE_JOUEUR':
      return {
        ...state,
        joueurs: [...state.joueurs, action.joueur],
        scores: [...state.scores, scoreVierge(action.joueur)],
        peutDemarrer: state.joueurs.length > 0
      }

    case 'DEMARRER_PARTIE':
      return {
        ...state,
        phase: 'EN_COURS'
      }

    case 'LANCER_FLECHETTE':
      const lancer = {
        lanceur: action.joueur,
        chiffre: action.chiffre,
        touches: action.touches
      }
      const nouveauScore = calculerLeNouveauScore(state.scores, lancer)
      const vainqueursDuNouveauScore = vainqueurs(nouveauScore)

      return {
        ...state,
        scores: nouveauScore,
        vainqueurs: vainqueursDuNouveauScore,
        phase: vainqueursDuNouveauScore.length > 0 ? 'TERMINEE' : state.phase
      }


    default:
      return state
  }
}