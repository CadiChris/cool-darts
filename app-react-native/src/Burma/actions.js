import Contrats from "./arbitrage/Contrats";

export const DEMARRER_PARTIE = "DEMARRER_PARTIE";
export const demarrerPartie = joueurs => ({
  type: DEMARRER_PARTIE,
  joueurs
});

export const VOLEE = "VOLEE";
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches)
  }
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches)
  }
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches)
  }
});

export const voleeSurBull = (lanceur, nombreDeBulls) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.bull(nombreDeBulls)
  }
});
