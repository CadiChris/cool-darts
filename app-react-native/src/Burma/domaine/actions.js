import Contrats from './arbitrage/Contrats';

export const DEMARRER_BURMA = 'BURMA/DEMARRER';
export const demarrerBurma = joueurs => ({
  type: DEMARRER_BURMA,
  joueurs,
});

export const VOLEE = 'BURMA/VOLEE';
export const voleeSurChiffre = (lanceur, chiffre, nombreDeTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.chiffre(chiffre, nombreDeTouches),
  },
});

export const voleeSurDouble = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.double(chiffresTouches),
  },
});

export const voleeSurTriple = (lanceur, chiffresTouches) => ({
  type: VOLEE,
  payload: {
    lanceur,
    contrat: new Contrats.triple(chiffresTouches),
  },
});
