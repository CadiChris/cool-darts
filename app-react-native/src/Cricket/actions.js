export const demarrerPartie = joueurs => ({
  type: "DEMARRER_PARTIE",
  joueurs
});

export const lancerFlechette = (joueur, chiffre, touches) => ({
  type: "LANCER_FLECHETTE",
  joueur,
  chiffre,
  touches
});

export const nouvellePartie = () => ({
  type: "NOUVELLE_PARTIE"
});
