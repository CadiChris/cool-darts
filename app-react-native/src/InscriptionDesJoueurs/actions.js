export const INSCRIRE_JOUEUR = "INSCRIRE_JOUEUR";
export const inscrireJoueur = nomDuJoueur => ({
  type: INSCRIRE_JOUEUR,
  nomDuJoueur
});

export const DESINSCRIRE_JOUEUR = "DESINSCRIRE_JOUEUR";
export const desinscrireJoueur = nomDuJouer => ({
  type: DESINSCRIRE_JOUEUR,
  nomDuJouer
});

export const CHOISIR_JEU = "CHOISIR_JEU";
export const choisirJeu = jeuChoisi => ({
  type: CHOISIR_JEU,
  jeuChoisi
});
