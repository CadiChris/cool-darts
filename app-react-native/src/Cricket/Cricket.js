import React, { Component } from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import { connect } from "react-redux";
import TableauDesScores from "./TableauDesScores/TableauDesScores.container";
import CommandesDeLaPartie from "./CommandesDeLaPartie/CommandesDeLaPartie.container";
import { demarrerPartie } from "./actions";

class Cricket extends Component {
  componentDidMount() {
    const { dispatch, joueurs } = this.props;
    dispatch(demarrerPartie(joueurs));
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <View style={{ flexGrow: 1 }}>
          <TableauDesScores />
        </View>
        <CommandesDeLaPartie />
      </View>
    );
  }
}

Cricket.propTypes = {
  joueurs: PropTypes.array.isRequired
};

export default connect()(Cricket);
