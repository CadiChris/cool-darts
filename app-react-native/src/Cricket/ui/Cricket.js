import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {connect} from 'react-redux';
import KeepAwake from 'react-native-keep-awake';
import TableauDesScores from './TableauDesScores/TableauDesScores.container';
import {demarrerCricket} from '../domaine/actions';

class Cricket extends Component {
  componentDidMount() {
    const {dispatch, joueurs} = this.props;
    dispatch(demarrerCricket(joueurs));
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <TableauDesScores />
        <KeepAwake />
      </View>
    );
  }
}

Cricket.propTypes = {
  joueurs: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  joueurs: state.inscriptionDesJoueurs.inscrits,
});

export default connect(mapStateToProps)(Cricket);
