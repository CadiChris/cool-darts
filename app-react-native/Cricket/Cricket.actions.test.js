import { inscrireJoueur, demarrerPartie } from './Cricket.actions'

it('inscrit un joueur', () => {
  expect(inscrireJoueur('J1')).toMatchSnapshot()
})

it('dÃ©marre la partie', () => {
  expect(demarrerPartie()).toMatchSnapshot()
})