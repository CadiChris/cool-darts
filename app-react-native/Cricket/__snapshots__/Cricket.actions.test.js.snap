// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`d√©marre la partie 1`] = `
Object {
  "type": "DEMARRER_PARTIE",
}
`;

exports[`inscrit un joueur 1`] = `
Object {
  "joueur": "J1",
  "type": "INSCRIRE_JOUEUR",
}
`;
